src/inmanta/parser/plyInmantaParser.py

```diff
@@ -368,6 +376,20 @@ def p_attr_undef(p: YaccProduction) -> None:
     attach_from_string(p, 2)
 
 
+def p_attr_dict_err(p: YaccProduction) -> None:
+    """attr : DICT CID empty
+    | DICT CID '=' map_def
+    | DICT CID '=' NULL
```

Comment:
```suggestion
    | DICT empty CID '=' map_def
    | DICT empty CID '=' NULL
```
I think this should allow you to use `p[3]` for all scenarios.

Reply:
Neat trick